### DISCUSSION TITLE
Add custom data to INVITE header

### DISCUSSION THREAD

**Discussion by AlexTaoSaab** [sipsorcery-org/sipsorcery#531] [Q&A]
How can I add some custom data to INVITE headers when I create a call? Just need to add something simple like one or two key:value strings. 

Best regards,
Alex

**Comment by sipsorcery** [comment]
Use the `SIPCallDescriptor` overload when placing a call and add your headers to [SIPCallDescriptor.CustomHeaders](https://github.com/sipsorcery-org/sipsorcery/blob/1c75694a7ade416bc89f7f420bddbb54de5fc87b/src/app/SIPUserAgents/SIPCallDescriptor.cs#L96).

**Reply by AlexTaoSaab** [reply]
Thanks for the quick answer.

It seems that it does get sent out, but for some reason, the receiever does not get the header. Checked the sip headers via wireshark and it's not there either. Seems to get lost on the way somewhere.

**Comment by sipsorcery** [comment]
````
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using SIPSorcery.SIP;
using SIPSorcery.SIP.App;
using SIPSorcery.Media;
using SIPSorceryMedia.Windows;

namespace SIPGetStarted
{
    class Program
    {
        private static string DESTINATION = "sip:helloworld@sipsorcery.cloud";

        static async Task Main()
        {
            Console.WriteLine("SIP Get Started");

            var userAgent = new SIPUserAgent();
            var winAudio = new WindowsAudioEndPoint(new AudioEncoder());
            var voipMediaSession = new VoIPMediaSession(winAudio.ToMediaEndPoints());
            voipMediaSession.AcceptRtpFromAny = true;

            var customHeaders = new List<string> { "Custom1: 1234", "Custom2: 4567" };
            var callDest = new SIPCallDescriptor(null, null, DESTINATION, null, null, null, customHeaders, null, SIPCallDirection.Out, null, null, null);
            bool callResult = await userAgent.Call(callDest, voipMediaSession);
            Console.WriteLine($"Call result {((callResult) ? "success" : "failure")}.");

            Console.WriteLine("Press any key to hangup and exit.");
            Console.ReadLine();
        }
    }
}
````

![sipcustom](https://user-images.githubusercontent.com/197660/126230797-9efda295-fd2e-470d-a215-11c7bdecb786.png)

**Reply by AlexTaoSaab** [reply]
I did some more digging and I had hoped to be able to send a custom header from one sip client to the other. As I mentioned in my last post, the sip header is being sent out from the caller correctly. But my Asterisk media server does not propagate the header to the callee. So the neither the solution nor problem lies in Sip sorcery. I've solved my problem in some other way for now. Still, appreciate the help!
